src/components/SummaryConfirmation/index.js

```diff
@@ -1,55 +1,113 @@
 import {Literal} from 'components/Literal';
 import Loader from 'components/Loader';
-import PrivacyCheckbox from 'components/PrivacyCheckbox';
 import {Toolbar, ToolbarList} from 'components/Toolbar';
 import {SUBMISSION_ALLOWED} from 'components/constants';
 
-export const PRIVACY_POLICY_ENDPOINT = 'config/privacy_policy_info';
+export const PRIVACY_POLICY_ENDPOINT = 'config/declarations_info_list';
 
-const getPrivacyPolicyInfo = async baseUrl => {
+const getDeclarationsInfo = async baseUrl => {
   return await get(`${baseUrl}${PRIVACY_POLICY_ENDPOINT}`);
 };
 
+const isSubmitEnabled = (loading, declarationsInfo = [], declarationsValues) => {
+  if (loading) return false;
+
+  return declarationsInfo.every(info => {
+    if (!info.validate.required) return true;
+
+    return Boolean(declarationsValues[info.key]);
+  });
+};
+
+const getDeclarationValues = (declarationsInfo = [], formikValues) => {
+  const declarationValuesAsArray = declarationsInfo
+    .map(info => {
+      if (!info.validate.required) return null;
+      return [info.key, formikValues[info.key] || false];
+    })
+    .filter(item => Array.isArray(item));
+
+  return Object.fromEntries(declarationValuesAsArray);
+};
+
+const getInitialWarningValues = declarationsValues => {
+  const warningValuesAsArray = Object.entries(declarationsValues).map(([declarationKey]) => [
+    declarationKey,
+    false,
+  ]);
+  return Object.fromEntries(warningValuesAsArray);
+};
+
 const SummaryConfirmation = ({submissionAllowed, onPrevPage}) => {
   const {baseUrl} = useContext(ConfigContext);
   const canSubmit = submissionAllowed === SUBMISSION_ALLOWED.yes;
 
   const {
     loading,
-    value: privacyPolicyConfig = {},
+    value: declarationsInfo = [],
     error,
   } = useAsync(async () => {
-    if (!canSubmit) return undefined;
-    return await getPrivacyPolicyInfo(baseUrl);
-  }, [baseUrl, getPrivacyPolicyInfo, canSubmit]);
+    if (!canSubmit) return [];
+
+    return await getDeclarationsInfo(baseUrl);
+  }, [baseUrl, getDeclarationsInfo, canSubmit]);
+  const {values: formikValues} = useFormikContext();
+
   if (error) throw error;
 
-  const {requiresPrivacyConsent, privacyLabel} = privacyPolicyConfig;
-  const {
-    values: {privacy: policyAccepted},
-  } = useFormikContext();
+  const submitDisabled = !isSubmitEnabled(loading, declarationsInfo, formikValues);
+
+  const [declarationsWarnings, setDeclarationWarnings] = useState(
+    getInitialWarningValues(getDeclarationValues(declarationsInfo, formikValues))
+  );
+
+  const showWarnings = formikValues => {
+    const declarationValues = getDeclarationValues(declarationsInfo, formikValues);
+
+    let updatedWarnings = {...declarationsWarnings};
+    Object.entries(declarationValues).forEach(([declarationKey, declarationValue]) => {
+      if (!declarationValue) {
+        updatedWarnings = {...updatedWarnings, [declarationKey]: true};
+      }
+    });
+    setDeclarationWarnings(updatedWarnings);
```

Comment:
perhaps a reducer is more readable here? now it looks like we're `zip`'ing arrays to match field name + value + showWarning yes/no while we can probably make that more self-contained?